c/misra/src/rules/RULE-5-1/ExternalIdentifiersNotDistinct.ql

```diff
@@ -0,0 +1,51 @@
+/**
+ * @id c/misra/external-identifiers-not-distinct
+ * @name RULE-5-1: External identifiers shall be distinct
+ * @description Using nondistinct external identifiers results in undefined behaviour.
+ * @kind problem
+ * @precision very-high
+ * @problem.severity warning
+ * @tags external/misra/id/rule-5-1
+ *       correctness
+ *       maintainability
+ *       readability
+ *       external/misra/obligation/required
+ */
+
+import cpp
+import codingstandards.c.misra
+import codingstandards.cpp.Linkage
+
+class ExternalIdentifiers extends Declaration {
+  ExternalIdentifiers() {
+    this.getName().length() > 31 and
```

Comment:
```suggestion
    this.getName().length() >= 31 and
```
I think we want to capture the case where one identifier is exactly 31 characters long, and another is more than 31, but both match in the first 31 characters. Suggest adding a test case for that as well.

Reply:
done